MapboxVisionTests/RecordingQuotaTests.swift

```diff
@@ -0,0 +1,124 @@
+        // Given
+        let memoryZeroBytes: Byte = 0
+
+        // When // Then
+        XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryZeroBytes))
+    }
+
+    func testReserveMethodDoesNotThrowIfMemoryQuotaWasNotExceeded() {
+        // Given
+        let memoryToReserveLessThanQuota = initialMemoryQuota - 1
+
+        // When // Then
+        XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveLessThanQuota))
+    }
+
+    func testReserveMethodAllowsToReserveMemorySeveralTimesIfMemoryQuotaWasNotExceeded() {
+        // Given
+        let memoryToReserveAtOneTime = 1 * kByte
+        let expectedNumberOfTimesWeCanReserveMemory = initialMemoryQuota / memoryToReserveAtOneTime
+
+        // When
+        for _ in 1...expectedNumberOfTimesWeCanReserveMemory {
+            // Then
+            XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveAtOneTime))
+        }
+    }
+
+    func testReserveMethodThrowsAfterSeveralReservationsButOnlyIfMemoryQuotaExceeded() {
+        // Given
+        let memoryToReserveAtOneTime = 1 * kByte
+        let expectedNumberOfTimesWeCanReserveMemory = initialMemoryQuota / memoryToReserveAtOneTime
+
+        for _ in 1...expectedNumberOfTimesWeCanReserveMemory {
+            XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveAtOneTime))
+        }
+
+        // When // Then
+        XCTAssertThrowsError(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveAtOneTime))
+    }
+
+    func testReserveMethodThrowsIfReservesMemoryMoreThanQuota() {
+        // Given
+        let memoryToReserveExceedingQuota = initialMemoryQuota + 1
+
+        // When // Then
+        XCTAssertThrowsError(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveExceedingQuota))
+    }
+
+    func testReserveMethodDoesNotThrowIfCurrentQuotaWasReset() {
+        // Given
+        let memoryToReserveMoreThanHalfAQuota = (initialMemoryQuota / 2) + 1
+        XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveMoreThanHalfAQuota))
+
+        // When
+        Thread.sleep(forTimeInterval: refreshInterval)
+
+        // Then
+        XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveMoreThanHalfAQuota))
+    }
+
+    func testCurrentQuotaFullyResetsWhenRefreshIntervalIsTimedOut() {
+        // Given
+        let memoryToReserveEqualsToQuota = initialMemoryQuota!
+        XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveEqualsToQuota))
+
+        // When
+        Thread.sleep(forTimeInterval: refreshInterval)
+
+        // Then
+        XCTAssertNoThrow(_ = try recordingQuota.reserve(memoryToReserve: memoryToReserveEqualsToQuota))
+    }
+
+    // MARK: - Private functions
+
+    func resetUserDefaultsState() {
+        UserDefaults.standard.removeObject(forKey: "lastResetTimeKey")
```

Comment:
Maybe it's better to make `RecordingQuota.Keys` internal for this purpose?

Reply:
Sure, thank you! :)